crates/provider/src/kubernetes/namespace.rs

```diff
@@ -121,15 +142,44 @@ where
             .write(dest_path, serialized_manifest)
             .await?;
 
+        // Ensure namespace isolation and minimal resources IFF we are running in CI
+        if env::var("RUN_IN_CI").unwrap_or_default() == "1" {
+            let np_manifest = helpers::apply_replacements(
+                include_str!("./static-configs/namespace-network-policy.yaml"),
+                &HashMap::from([("namespace", self.name())]),
+            );
+
+            // Apply NetworkPolicy manifest
+            self.k8s_client
+                .create_static_resource(&self.name, &np_manifest)
+                .await
+                .map_err(|err| {
+                    ProviderError::CreateNamespaceFailed(self.name.to_string(), err.into())
+                })?;
+
+            // Apply LimitRange manifest
+            self.k8s_client
+                .create_static_resource(
+                    &self.name,
+                    include_str!("./static-configs/baseline-resources.yaml"),
+                )
+                .await
+                .map_err(|err| {
+                    ProviderError::CreateNamespaceFailed(self.name.to_string(), err.into())
+                })?;
+        }
```

Comment:
Here, from what I understand, we deploy static resources if we are running in CI, shouldn't be the contrary ? To not deploy them in CI ?

Also, we can extract a function ```initialize_static_resources``` to avoid cluttering the ```initialize_k8s```:


Something like this:
```suggestion
        if helpers::running_in_ci() {
            self.initialize_static_resources().await?;
        }
```

```rust
mod helpers {
    pub fn running_in_ci() -> bool {
        env::var("RUN_IN_CI").unwrap_or_default() == "1"
    }
}
```

Reply:
No, we need to deploy those manifest only in `CI` to ensure `namespace` isolation (this could be deployed always) and the baseline resources should be deployed only for `CI` to ensure the minimal resource allocation.